_extensions/software-licensing/change-license-expiration-length.html

```diff
@@ -0,0 +1,25 @@
+---
+layout: snippet
+title: Changing License Expiration Length
+description: With Software Licensing the license length (time to expire) can be easily changed via a simple filter. This example snippet will illustrate how to change the length of licenses, and also show how to conditionally change it based on the product that the license is being generated for.
+collection: extensions
+category: software-licensing
+---
+
+<?php
+/**
+ * Plugin Name: Easy Digital Downloads - Change License Expiration Length
+ * Description: Changes the length of licenses
+ * Author: Easy Digital Downloads
+ * Author URI: https://easydigitaldownloads.com/
+ * Version: 1.0
+ */
+
+function pw_edd_license_length( $length, $payment_id, $download_id, $license_id ) {
+	
+	if( $download_id == 14 )
+		return '+2 year'; // give a 2 year expiration for product ID 14
+
+	return '+1 year'; // this is default
```

Comment:
It's possible I don't understand the goal of the snippet, but it makes more sense to me that this would be:

```php
return $length
```

As it is, this snippet will make **all durations** `+1 year`, regardless of settings, except for download ID 14, which is set to `+2 year`. So everything is always 1 or 2 years and all UI settings are ignored completely.

I'd have thought what we'd want to do is use the existing product settings (which is in `$length`) and only make changes for download ID 14.